### PR TITLE
fix(pointcloud_preprocessor): ignore NaN points in crop box filter

### PR DESCRIPTION
**Pull request by VRichardJP**
## Description

See https://github.com/autowarefoundation/autoware.universe/issues/3112#issuecomment-1475480939
TL;DR: it is dangerous to keep NaN values around, it's better to ignore them than to try to "fix" them.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
@sykwer Can you review this?

**Comment by VRichardJP** [comment]
Note, that this PR also includes the fix from https://github.com/autowarefoundation/autoware.universe/pull/3122#issuecomment-1475991255
So the other PR may be discarded if the present one is accepted.

**Review by mitsudome-r — APPROVED** [review]
LGTM, but we need approval from package maintainer before merging. 

@yukkysaito @amc-nu @miursh

**Review by yukkysaito — APPROVED** [review]
LGTM

**Review by miursh — APPROVED** [review]

**Comment by miursh** [comment]
@VRichardJP Could you fix the conflicts?

**Comment by VRichardJP** [comment]
@miursh Done!
